### DISCUSSION TITLE
Animation Optimization Ideas

### DISCUSSION THREAD

**Discussion by Xavman42** [DigiScore/neoscore#90] [Ideas]
I'm looking into ways to increase the frame-rate in my animated score. Some of this involves refactoring loops and maybe implementing something like cython or numba, but the real bottleneck I'm running into seems to come down to every object in the document being re-rendered every frame. This can be mitigated quite a bit by manually culling objects when they aren't in the scene - it turns out that qt does not cull things outside the viewport for you.

In my score, objects are static 99% of the time, and I pan the viewport to make it seem like things are moving. I wonder if it would be possible to create another version of `neoscore.core.neoscore.wrapped_refresh_func` where I can manually call `document.render` whenever I want to instantiate objects in the document - some function like a `neoscore.update_render()`. As far as I can tell, setting the pan and zoom of the viewport does not require a document re-render, so I think I should be able to make things appear like they're moving without re-rendering everything every frame.

**Comment by ajyoon** [comment]
That should be straightforward, I'll look into it this week.

**Comment by ajyoon** [comment]
This is released now in 0.1.11
